src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/ApproverCaseloadServiceTest.kt

```diff
@@ -41,359 +41,1030 @@ class ApproverCaseloadServiceTest {
       )
+      whenever(prisonerSearchApiClient.searchPrisonersByNomisIds(listOf(nomisId))).thenReturn(listOf(aPrisonerSearchResult))
+      whenever(communityApiClient.getStaffDetailsByUsername(comUsernames)).thenReturn(listOf(aUser))
 
       val approvalCases = service.getApprovalNeeded(aListOfPrisonCodes)
 
-      assertThat(approvalCases).hasSize(2)
+      assertThat(approvalCases).hasSize(1)
 
       with(approvalCases.first()) {
-        assertThat(licenceId).isEqualTo(1L)
-        assertThat(name).isEqualTo("Bob Mortimar")
-        assertThat(prisonerNumber).isEqualTo("A1234AA")
-        assertThat(submittedByFullName).isEqualTo("X Y")
-        assertThat(releaseDate).isEqualTo((LocalDate.of(2021, 10, 22)))
-        assertThat(urgentApproval).isFalse()
-        assertThat(approvedBy).isEqualTo("jim smith")
-        assertThat(approvedOn).isEqualTo((LocalDateTime.of(2023, 9, 19, 16, 38, 42)))
-        assertThat(isDueForEarlyRelease).isFalse()
-        with(probationPractitioner!!) {
-          assertThat(staffCode).isEqualTo("AB012C")
-          assertThat(name).isEqualTo("Test Test")
-        }
-      }
-
-      with(approvalCases.last()) {
-        assertThat(licenceId).isEqualTo(2L)
-        assertThat(name).isEqualTo("John Doe")
-        assertThat(prisonerNumber).isEqualTo("B1234BB")
-        assertThat(submittedByFullName).isEqualTo("X Y")
-        assertThat(releaseDate).isEqualTo((LocalDate.of(2021, 10, 22)))
-        assertThat(urgentApproval).isFalse()
-        assertThat(approvedBy).isEqualTo("jim smith")
-        assertThat(approvedOn).isEqualTo((LocalDateTime.of(2023, 9, 19, 16, 38, 42)))
-        assertThat(isDueForEarlyRelease).isFalse()
-        with(probationPractitioner!!) {
-          assertThat(staffCode).isEqualTo("DE012F")
-          assertThat(name).isEqualTo("Test2 Test2")
-        }
+        assertThat(releaseDate).isEqualTo((LocalDate.of(2024, 6, 20)))
       }
 
       verify(prisonApproverService, times(1)).getLicencesForApproval(aListOfPrisonCodes)
-      verify(probationSearchApiClient, times(1)).searchForPeopleByNomsNumber(nomisIds)
-      verify(prisonerSearchApiClient, times(1)).searchPrisonersByNomisIds(listOf("A1234AA", "B1234BB"))
+      verify(probationSearchApiClient, times(1)).searchForPeopleByNomsNumber(listOf(nomisId))
+      verify(prisonerSearchApiClient, times(1)).searchPrisonersByNomisIds(listOf(nomisId))
       verify(communityApiClient, times(1)).getStaffDetailsByUsername(comUsernames)
     }
+  }
 
-    @Test
-    fun `the caseload is sorted correctly`() {
-      val nomisIds = listOf("A1234AA", "B1234BB", "C1234CC")
-      val comUsernames = listOf("smills", "jdoe")
+  @Nested
+  inner class `Build getRecentlyApproved caseload`() {
 
-      whenever(prisonApproverService.getLicencesForApproval(aListOfPrisonCodes)).thenReturn(
-        listOf(
-          aLicenceSummaryApproverView.copy(
-            actualReleaseDate = LocalDate.of(2024, 6, 21),
+    @Nested
```

Comment:
<!-- Notes are always non-blocking and simply highlight something the reader should take note of. -->
**note:** ‏Reformatting above, new tests start here